### PR TITLE
Modified to encrypt the keychain password.

### PR DESCRIPTION
**Pull request by kazuhidet**
- Modified to save the keychain password encrypted in 'Configure system' and 'Job configuration'.
- The steps to unlock the macOS keychain were separated so that they could be used alone.

### CODE REVIEW

**Comment by kazuhidet** [comment]
I think that this PR should be merged early, as I think it should be fixed as soon as possible the problem that the keychain password was saved in plain text, how about it ?

It could not adversely affect anything else, at least as long as I check it myself.

File: src/main/java/au/com/rayh/Keychain.java [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
-    private String keychainPassword;
+    private Secret keychainPassword;
     private Boolean inSearchPath;
 
     public Keychain() {
     }
 
     @DataBoundConstructor
-    public Keychain(String keychainName, String keychainPath, String keychainPassword, Boolean inSearchPath) {
+    public Keychain(String keychainName, String keychainPath, Secret keychainPassword, Boolean inSearchPath) {
```
Comment (mat1e) [inline src/main/java/au/com/rayh/Keychain.java]: Is it not better to add an other default constructor like "public Keychain()" to use @DataBoundSetter? Like this you can deprecate this one and modify the setting of keychainPassword adding Secret.fromString().

Comment (mat1e) [inline src/main/java/au/com/rayh/Keychain.java]: Is it not better to add an other default constructor like "public Keychain()" to use @DataBoundSetter? Like this you can deprecate this one and modify the setting of keychainPassword adding Secret.fromString().

**Comment by kazuhidet** [comment]
Jenkins official document "Writing Pipeline-Compatible Plugins" say "Instead you should integrate with the Credentials plugin."
https://jenkins.io/doc/developer/plugin-development/pipeline-integration/

I think this mean Information about authentication had better do it handled through "credential plugin" rather than stored by plugin itself.

What do you think, guys?

**Pull request merged**